Remora.Rest.Core/Optional.cs

```diff
@@ -155,4 +183,86 @@ public override string ToString()
             ? $"{{{_value?.ToString() ?? "null"}}}"
             : "Empty";
     }
+
+    /// <summary>
+    /// Returns the value of the current <see cref="Optional{TValue}"/>, or <c>default</c> if the current
+    /// <see cref="Optional{TValue}"/> is empty.
+    /// </summary>
+    /// <returns>The value of <c>this</c> or <c>default</c> if none is present.</returns>
+    public TValue? OrDefault()
+    {
+        return TryGet(out var value) ? value : default;
+    }
+
+    /// <summary>
+    /// Returns the value of the current <see cref="Optional{TValue}"/>, or <paramref name="defaultValue"/> if
+    /// <see cref="Optional{TValue}"/> is <b>null or empty</b>.
+    /// </summary>
+    /// <param name="defaultValue">The default value to fallback to if the optional is empty.</param>
+    /// <returns>
+    /// The value of <see cref="Optional{TValue}"/> or <paramref name="defaultValue"/> if none is present.
+    /// </returns>
+    [return: NotNullIfNotNull(nameof(defaultValue))]
+    public TValue? OrDefault(TValue? defaultValue)
+    {
+        return IsDefined(out var value) ? value : defaultValue;
+    }
+
+    /// <summary>
+    /// Returns the value of the current <see cref="Optional{TValue}"/>, or throws the exception in
+    /// <paramref name="func"/> if empty.
+    /// </summary>
+    /// <param name="func">The function generating an <see cref="Exception"/>.</param>
+    /// <returns>The value of <see cref="Optional{TValue}"/>.</returns>
+    /// <exception cref="Exception">If <see cref="Optional{TValue}"/> is empty.</exception>
+    public TValue OrThrow([RequireStaticDelegate] Func<Exception> func)
```

Comment:
A remark about why this is a `Func` and why it's supposed to be a static delegate might be in order, since it's a non-obvious optimization.